src/main/java/org/broadinstitute/hellbender/utils/iterators/ByteArrayIterator.java

```diff
@@ -9,18 +9,34 @@
 public final class ByteArrayIterator implements Iterator<Byte> {
     private final byte[] byteArray;
     private int currentPosition;
+    private int stopIndex;
 
     public ByteArrayIterator( final byte[] byteArray ) {
+        this(byteArray, 0, byteArray.length);
+    }
+
+    /**
+     * Iterates through this array, starting at index 'firstIndex' and going until
+     * (but not including) index 'stopIndex'.
+     */
+    public ByteArrayIterator( final byte[] byteArray, int firstIndex, int stopIndex ) {
+        if (firstIndex<0) {
+            throw new IllegalArgumentException("firstIndex should be at least 0, but it is " + firstIndex);
+        }
```

Comment:
Also check that `firstIndex` < byteArray.length -- can just call `Utils.validIndex(firstIndex, byteArray.length);`